tests/scripts/all.sh

```diff
@@ -67,42 +67,23 @@
 # Notes for maintainers
 # ---------------------
 #
-# The bulk of the code is organized into functions that follow one of the
+# The code is organized into functions that follow one of the
 # following naming conventions:
 #  * pre_XXX: things to do before running the tests, in order.
-#  * component_XXX: independent components. They can be run in any order.
-#      * component_check_XXX: quick tests that aren't worth parallelizing.
-#      * component_build_XXX: build things but don't run them.
-#      * component_test_XXX: build and test.
-#      * component_release_XXX: tests that the CI should skip during PR testing.
-#  * support_XXX: if support_XXX exists and returns false then
-#    component_XXX is not run by default.
 #  * post_XXX: things to do after running the tests.
 #  * other: miscellaneous support functions.
 #
-# Each component must start by invoking `msg` with a short informative message.
+# The all.sh script has been split up into two main sections.
+# all.sh contains the framework that is required for executing the
```

Comment:
Please also split out the “framework” part into another file that only defines functions and doesn't run them, so that we can create other entry points that do something like
```
#!/bin/bash
# Convenience script to test all configurations
. runner-framework.sh
. components-configurations.sh
runner
```